### ISSUE TITLE
[Products] Subscription price fields showing for all products

### ISSUE THREAD

**Issue by rachelmcr** [woocommerce/woocommerce-ios#13569]
**Describe the bug**

When editing a product’s price on a store with the [Subscriptions extension](https://woocommerce.com/products/woocommerce-subscriptions/), I see the subscription product price fields (e.g. Billing interval, sign-up fee) even if the product is not a subscription product.

**To Reproduce**
1. Select a store with the Subscriptions extension.
2. Go to the Products tab.
3. Select any product.
4. Select `Price`.
5. Notice that the subscription price fields (`Billing interval`, `Sign-up Fee`) appear for all products.

**Screenshots**

Product details|Price
-|-
![image](https://github.com/user-attachments/assets/be0ead23-7741-49e1-b045-3fdc004a7690)|![image](https://github.com/user-attachments/assets/e54e388f-172f-4dc2-a9ad-824946d04030)

**Expected behavior**

I expect the subscription price fields to only appear for subscription products.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by rachelmcr** [comment]
I realized this is happening because my test store also has the [All Products For Woo Subscriptions](https://woocommerce.com/products/all-products-for-woocommerce-subscriptions/) extension. This makes it possible to turn any product into a subscription product.

We don't fully support this extension in the app, so it doesn't quite work as expected (this extension has additional product settings for enabling subscriptions). I also tried disabling the extension, but it doesn't resolve the issue for this store — it looks like the metadata we use for subscriptions in the app is still saved on all the products, so those fields are shown in the app.

That said, I tested again on a store that only had the Subscription extension, and it works as expected there. So until we decide to fully support All Products For Woo Subscriptions, I think this is acceptable behavior.
